### PR TITLE
Small PHPDocs fixes [Catalog module]

### PR DESCRIPTION
**Pull request by ghost**
### Description
- Deleted unused code
- Small PHPDocs fixes in Catalog module

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/Api/CategoryLinkRepositoryInterface.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ public function delete(\Magento\Catalog\Api\Data\CategoryProductLinkInterface $p
     /**
      * Remove the product assignment from the category by category id and sku
      *
-     * @param string $categoryId
+     * @param int $categoryId
```
Comment (ghost) [inline app/code/Magento/Catalog/Api/CategoryLinkRepositoryInterface.php:42]: see app/code/Magento/Catalog/Model/CategoryLinkManagement.php:88

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3916](https://jira.corp.magento.com/browse/ENGCOM-3916) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Comment by magento-engcom-team** [comment]
Hi @SikailoISM. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Pull request merged**